crates/telio-firewall/src/firewall.rs

```diff
@@ -572,22 +621,19 @@ impl StatefullFirewall {
         }
 
         // no value in cache, insert and allow only if ip is whitelisted
-        if !whitelist.is_port_whitelisted(peer, key.local_port) {
+        if !is_port_whitelisted(key.link.local_port) {
             telio_log_trace!("Dropping UDP packet {:?} {:?}", key, peer);
             return false;
         }
 
-        let conninfo = UdpConnectionInfo {
-            is_remote_initiated: true,
-        };
-
-        telio_log_trace!(
-            "Updating UDP conntrack entry {:?} {:?} {:?}",
-            key,
-            peer,
-            conninfo
-        );
-        udp_cache.insert(key, conninfo);
+        telio_log_trace!("Updating UDP conntrack entry {:?} {:?}", key, peer,);
+        match udp_cache.entry(key) {
+            Entry::Occupied(mut occ) => occ.get_mut().is_remote_initiated = true,
```

Comment:
It seems like this `Entry::Occupied` ARM cannot really be executed ever, as if the if statement at line 607 was false (otherwise it would have returned by here), means, that it must be Vacant. 

Maybe it would be clearer to convert that `if` into `match` statement, to avoid such an arm?

Reply:
Actually, I cannot replace it with entry altogether since `entry()` does not invalidate outdated values

Reply:
Or I will try to change the `entry()` API so that it invalidates the keys

Reply:
I made the changes, but I'm leaving this discussion open as I'm not sure if that's the correct approach